File: src/main/com/stockcomp/util/Metrics.java
package com.stockcomp.util;

import io.micrometer.core.instrument.Counter;
import io.micrometer.core.instrument.MeterRegistry;
import org.springframework.stereotype.Component;

@Component
public class Metrics {

    public final Counter SIGN_UP_USER;

    private Metrics(MeterRegistry meterRegistry) {
        SIGN_UP_USER = meterRegistry.counter("signed.up.user");
    }

}

File: src/main/com/stockcomp/util/Metrics.kt
package com.stockcomp.util

import io.micrometer.core.instrument.Counter
import io.micrometer.core.instrument.MeterRegistry
import org.springframework.stereotype.Component

@Component
class Metrics private constructor(meterRegistry: MeterRegistry) {

    val SIGN_UP_USER: Counter = meterRegistry.counter("signed.up.user")

}